# Written and placed in public domain by Jeffrey Walton.
# This patch fixes some issues with Flex.
#--- configure.ac
#+++ configure.ac
#@@ -28,7 +28,7 @@
# AC_CONFIG_SRCDIR([src/scan.l])
# AC_CONFIG_AUX_DIR([build-aux])
# LT_INIT
#-AM_INIT_AUTOMAKE([1.11.3 -Wno-portability foreign check-news std-options dist-lzip parallel-tests subdir-objects])
#+AM_INIT_AUTOMAKE([1.11.2 -Wno-portability foreign check-news std-options parallel-tests subdir-objects])
# AC_CONFIG_HEADER([src/config.h])
# AC_CONFIG_LIBOBJ_DIR([lib])
# AC_CONFIG_MACRO_DIR([m4])
--- src/misc.c
+++ src/misc.c
@@ -338,6 +338,10 @@
 
 void line_directive_out (FILE *output_file, int do_infile)
 {
+	/* Based on GCC object size checker */
+	#undef MAXLINE
+	# define MAXLINE (2048+64)
+
 	char    directive[MAXLINE], filename[MAXLINE];
 	char   *s1, *s2, *s3;
 	static const char line_fmt[] = "#line %d \"%s\"\n";
--- src/filter.c
+++ src/filter.c
@@ -336,6 +336,10 @@
  */
 int filter_fix_linedirs (struct filter *chain)
 {
+	/* Based on GCC object size checker */
+	#undef MAXLINE
+	# define MAXLINE (2048+64)
+
 	char   *buf;
 	const size_t readsz = 512;
 	int     lineno = 1;
